src/test/storage/partitioning/hash_function_test.cpp

```diff
@@ -0,0 +1,65 @@
+#include "../../base_test.hpp"
+#include "gtest/gtest.h"
+
+namespace opossum {
+
+class StorageHashFunctionTest : public BaseTest {
+ protected:
+  void SetUp() override {}
+};
+
+/*
+ * We want to make sure
+ *  1. that hashing the same value *always* produces the same result
+ *  2. that hashing different values *probably* produces different results
+ * 
+ * About 2 we can not be absolutely certains due to hash collisions. 
+ * There are no hash collisions for the used values, though.
+ * 
+ * We tests this for all data types AllTypeVariant can hold.
+ */
+
+TEST_F(StorageHashFunctionTest, HashInt32) {
+  HashFunction hf;
+  EXPECT_EQ(hf(INT32_C(42)), hf(INT32_C(42)));
+  EXPECT_EQ(hf(INT32_C(21)), hf(INT32_C(21)));
+  EXPECT_NE(hf(INT32_C(42)), hf(INT32_C(21)));
```

Comment:
the EXPECT_NE tests are not very useful. In general, i'm not sure how useful these tests are in general, because you are testing some random hash function that we don't control.

Reply:
We have replaced the hash function by boost::hash_value, which should use always the same hashing algorithm and seed.